Lib/idlelib/idle_test/test_parenclose.py

```diff
@@ -0,0 +1,97 @@
+'''Test idlelib.parenclose.
+'''
+from idlelib.ParenClose import ParenClose
+import unittest
+# Can't use mock text; it doesn't handle positioning correctly.
+from tkinter import Text
+from test.support import requires
+requires('gui')
+
+class MockEvent(object):
+    def __init__(self, char):
+        self.char = char
+
+
+class ParenCloseTest(unittest.TestCase):
+    def test_parenClose(self):
+        p = ParenClose()
+        p.text = Text()
+        p_open = p.p_open_event
+        p_close = p.p_close_event
+        t_open = p.t_open_event
+        back = p.back_event
+        delete = p.delete_event
```

Comment:
I do not like testing all 5 methods in one method.  On the other hand, I don't like repeating code, and using subtests allows multiple failures in one method.  A helper function might allow separate test functions without duplication, but this is a lower priority than getting good coverage to start with.